internal/pkg/service/buffer/definition/branch.go

```diff
@@ -0,0 +1,11 @@
+package definition
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/definition/key"
+)
+
+type Branch struct {
+	key.BranchKey
+	SoftDeletable
```

Comment:
`Branch` entity sluzi na to, aby som vedel povedat, ci bola branch uz inicializovana na pouzite so Stream API alebo nie.

Bez toho by som nevedel odlisit, neinicialitovanu branch od prazdnej.

Main branch sa vytvori (v dalsich PRs) automaticky pri prvom pouziti API.